src/Illuminate/Filesystem/FilesystemAdapter.php

```diff
@@ -842,6 +843,24 @@ public function deleteDirectory($directory)
         return true;
     }
 
+    /**
+     * Get the checksum for a file.
+     *
+     * @return string|false string checksum or false on error
+     *
+     * @throws UnableToProvideChecksum
+     */
+    public function checksum(string $path, array $options = []): string|false
+    {
+        try {
+            return $this->driver->checksum($path, $options);
+        } catch (UnableToProvideChecksum $e) {
+            throw_if($this->throwsExceptions(), $e);
+
+            return false;
```

Comment:
I'm also wondering why to return `false`? Wouldn't it be more convenient to return `null`?
This way you can also use `?string` return type instead of union.

Reply:
False is in line with the other failure responses when configured not to throw exceptions.